Sources/SwiftFormat/Rules/NoEmptyLineOpeningClosingBraces.swift

```diff
@@ -0,0 +1,130 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+@_spi(Rules)
+public final class NoEmptyLinesOpeningClosingBraces: SyntaxFormatRule {
```

Comment:
It's too bad we don't have a convenient way to just visit anything that conforms to `BracedSyntax` here.

Can you add `PrecedenceGroupDeclSyntax` and `SwitchExprSyntax`, which look like the only two other `BracedSyntax` node types that are missing here?

Reply:
To visit anything conforming to `BracedSyntax` you can do the following

```swift
public override func visitAny(_ node: Syntax) -> Syntax? {
  if let braced = node.as(BracedSyntax.self) {
    // do things
  } else {
    super.visitAny(node)
  }
}
```

Reply:
In general that would work for a regular visitor, but `generate-swift-format` scans the rules to figure out which concrete nodes they visit so that it can generate the pipeline visitor.

It might not be difficult to add `visitAny` support to the pipeline, but it would be better to do it separately from this PR.

Reply:
👍🏽 Just wanted to put this pattern out there because it’s not terribly obvious.

Reply:
I've wished more than once that `override func f(value: some T)` in a class would override any concrete method that matches.